### DISCUSSION TITLE
请教多WAN的情况下，应该如何写规则

### DISCUSSION THREAD

**Discussion by W3XA56Y3RDUKBS** [MetaCubeX/mihomo#1673] [Q&A]
情况说明：家中有两条宽带，中国电信和中国联通，我希望国内的域名都走联通的口出去

我进行了如下的设置：

proxies:

- name: "电信"
  type: direct
  udp: true
  ip-version: ipv6-prefer
  interface-name: eth0
  routing-mark: 1234

- name: "联通"
  type: direct
  udp: true
  ip-version: ipv6-prefer
  interface-name: eth1
  routing-mark: 2234

rules:
  - RULE-SET,CN,联通
  - RULE-SET,CNDomain,联通
  - GEOSITE,CN,联通
  - GEOIP,CN,联通

然而并不能用，所有国内的网站都无法访问， 想请教下各位应该如何设置？

**Comment by Plutonium141** [comment]
有必要用`routing-mark: `吗？

**Reply by W3XA56Y3RDUKBS** [reply]
哪怕删掉routing-mark，好像也是不行，不知道是不是我两个wan口都是pppoe拨号的缘故。请教下您知道应该如何创建两个走不同wan口的的direct节点嘛？

**Reply by Plutonium141** [reply]
我倒是没这么用过，是不是接口名弄错了？另外ip-version: ipv6-prefer最好是弄通之后再加

**Comment by htfcuddles** [comment]
我就是openwrt下多wan。interface-name或者routing-mark都可以，routing-mark更灵活可以HA可以Failover。不行的请排查iptables/nftable，停止mwan3等修改防火墙服务，cat /proc/net/nf_conntrack 看下mark=xxx正确与否（前提是restore过），clash只负责标记。路由正确与clash无关。两者需要pbr配合，通过ip rule和ip route show table x检查。
